cmd/node/factory/structs.go

```diff
@@ -232,43 +232,27 @@ func createTries(
 	trieContainer := state.NewDataTriesHolder()
 
 	trieFactoryArgs := factory.TrieFactoryArgs{
-		Cfg:                    args.config.AccountsTrieStorage,
 		EvictionWaitingListCfg: args.config.EvictionWaitingList,
 		SnapshotDbCfg:          args.config.TrieSnapshotDB,
 		Marshalizer:            marshalizer,
 		Hasher:                 hasher,
 		PathManager:            args.pathManager,
 		ShardId:                args.shardId,
-		PruningEnabled:         args.config.StateTrieConfig.PruningEnabled,
 	}
 	trieFactory, err := factory.NewTrieFactory(trieFactoryArgs)
 	if err != nil {
 		return nil, err
 	}
 
-	merkleTrie, err := trieFactory.Create()
+	merkleTrie, err := trieFactory.Create(args.config.AccountsTrieStorage, args.config.StateTrieConfig.PruningEnabled)
```

Comment:
I need to know if pruning is enabled before I create the trie. Should I build two separate functions for the trie factory (CreateWithPruning/CreateWithoutPruning) instead of Create()?

Reply:
Ok, no, you can leave it like this :)